### ISSUE TITLE
Move various scope properties into redux store

### ISSUE THREAD

**Issue by rmhowe** [medic/medic#5559]
As part of the ongoing changes to get ready to [move to Angular 7](https://github.com/medic/medic/issues/4750), this issue is for moving various scope properties into the redux store.

**AT:** Parts of the app touched include:
- All loading screens
- App "version" on the about page
- Error messages on analytics, contacts, messages, reports, tasks, and the edit user modal
- Url to the admin app
- The "facilities" filter on the reports page
- The currently selected document on analytics, contacts, messages, reports, and tasks

This is effectively a refactor with no new features added.

PR: https://github.com/medic/medic/pull/5644

**Comment by ngaruko** [comment]
The [PR](https://github.com/medic/medic/pull/5644) need some reviewed before we are able to merge @rmhowe .
https://github.com/medic/medic/pull/5644

**Comment by garethbowen** [comment]
Moving to 3.6.0 so it doesn't destabalise 3.5.0.

**Comment by rmhowe** [comment]
@ngaruko it's been reviewed but can't be merged until the `3.5.x` branch has been cut. Has it been ATed or were you waiting on the review?

**Comment by ngaruko** [comment]
Seeing this error @rmhowe  and page loading a wee slower that usual.
Did you want to have a look at this? Happened in Firefox
![image](https://user-images.githubusercontent.com/6979995/60410374-22e59780-9c1c-11e9-8818-de9d17e20582.png)
Also, the 'task' tab for admins normally shows an message along the line of `Tasks are disabled for admin users. If you need to see tasks, login as a normal user.` 
![image](https://user-images.githubusercontent.com/6979995/60411226-ec118080-9c1f-11e9-8d23-eba265c2d488.png)

But now it shows `no tasks`
![image](https://user-images.githubusercontent.com/6979995/60411054-2cbcca00-9c1f-11e9-89db-1a864ab8acc4.png)

**Comment by garethbowen** [comment]
I started to have a look at this but it's more serious than I hoped. Tasks is not working at all now - it always says "no tasks found" even on an offline user when there are some. This is caused by variables not being bound/passed correctly so the `$ctrl.tasksDisabled` and `$ctrl.hasTasks` are undefined.

Also, the other tabs are showing bugs saying "no x found" when there clearly are, eg:

![Screenshot from 2019-07-02 11-02-00](https://user-images.githubusercontent.com/2064938/60471496-08153080-9cb9-11e9-9c88-84556d3e19ac.png)
![Screenshot from 2019-07-02 11-02-11](https://user-images.githubusercontent.com/2064938/60471497-08153080-9cb9-11e9-9f39-92c22d73290d.png)

@rmhowe Could you please have another look at this PR and fix these issues? Also, could please rebase against master and resolve the conflicts so the AT will be close to what will actually be merged?  If you don't have time in the next couple of days, pleaselet me know and I'll have another look.

**Comment by rmhowe** [comment]
@garethbowen @ngaruko sorry got tripped up by Angular template syntax, too used to React. Should be fixed now.

**Comment by garethbowen** [comment]
@rmhowe Thanks Robbie! That was a fast turnaround.

**Comment by ngaruko** [comment]
There is another strange behavior observed here @rmhowe . On creating a new person, it takes a wee longer for the medic ID to be generated and in the meantime, if one deletes a contact (before the ID was generated), the UI shows that the the person has been deleted, but in actuality it is not.
See video attached.
[delete-contact.mov.zip](https://github.com/medic/medic/files/3352997/delete-contact.mov.zip)

**Comment by rmhowe** [comment]
@ngaruko I'm having trouble recreating the bug in my local environment. The video is great but were there some steps before the start of that video that led to the bug? (eg the creating a new person step) Would you be able to make a video that includes the whole process?

**Comment by ngaruko** [comment]
@rmhowe I think it is a really edge case scenario, not likely to happen in real life. Hard to reproduce even. But the steps were: create a new person, and delete it before the medic ID is generated. At first there was quite a delay and that's when it was not actually deleted, though there was a confirmation thereof in the UI. Now, I tried to replicate, and what I get is a error in the console and actually the `delete` button not doing anything.
![image](https://user-images.githubusercontent.com/6979995/60630015-48f17e80-9e4c-11e9-9267-4f9482ba7a44.png).

[create-delete-contact.mov.zip](https://github.com/medic/medic/files/3357075/create-delete-contact.mov.zip)

As I said it is really not likely to happen in production but I guess if we were to automate the UI, that could be and issue, but also it might indicate another flaw in the code. So, feel free to investigate and fix .

**Comment by ngaruko** [comment]
@rmhowe I realised the above is happening even on `master`, so not related to the refactoring. I will close this and raise a separate issue if needed. Thanks.

**Comment by rmhowe** [comment]
@ngaruko thanks for sorting that out. With respect to the facility filter I've pushed a fix to the branch, however I'm not sure what the expected behaviour is on master due to some peculiar code [here](https://github.com/medic/medic/blob/master/webapp/src/js/controllers/reports.js#L68) as a result of https://github.com/medic/medic/issues/4423, it's only allowing one facility to be displayed at a time? Regardless only showing one at a time appears to be the intended behaviour so then I believe my branch is behaving as expected.

**Comment by ngaruko** [comment]
Thanks @rmhowe There surely seems to be something weird with that feature. I will close this issue and investigate further on that and eventually raise a different issue for it!

**Comment by garethbowen** [comment]
Merged!
